Dockerfile

```diff
@@ -10,25 +10,21 @@
 # Supported architectures:
 # linux/amd64, linux/arm64, linux/arm/v7
 # https://github.com/docker-library/official-images#architectures-other-than-amd64
+FROM node:20-slim AS build
+WORKDIR /build
 
-FROM node:20-slim
-
-ENV NODE_ENV=production
-
-WORKDIR /usr/src/medplum
-
-# Add the application files
+# Add the application files: Docker handles extracting the tarball automatically
 ADD ./medplum-server.tar.gz ./
+# Install dependencies inside Docker container to handle multi-arch builds
+RUN npm ci --maxsockets 1
 
-# Install dependencies, create non-root user, and set permissions in one layer
-RUN npm ci --maxsockets 1 && \
-  groupadd -r medplum && \
-  useradd -r -g medplum medplum && \
-  chown -R medplum:medplum /usr/src/medplum
 
-EXPOSE 5000 8103
+FROM gcr.io/distroless/nodejs20-debian12:nonroot
```

Comment:
Hm.  Why did we need 2 phase build?  Does the distroless image not include `npm`?

Reply:
Correct: the distroless image is stripped down to the bare essentials to run the application, and specifically does not include the package manager